src/Raven.Client/Documents/Session/IAbstractDocumentQuery.cs

```diff
@@ -333,6 +333,8 @@ public interface IAbstractDocumentQuery<T>
         void AggregateUsing(string facetSetupDocumentId);
 
         void AddFromAliasToWhereTokens(string fromAlias);
+        
+        void AddFromAliasToOrderByTokens(string fromAlias);
```

Comment:
What about GroupBy tokens? Or Filter?

Reply:
I added aliasing for Filter. We only support `GroupBy` method for `Advanced.DocumentQuery` with field name string parameter, which forces using `SelectCount`, `SelectKey` and `SelectSum` for projection. It means we won't have alias from JS projection that we should handle.